iphone/Maps/UI/PlacePage/PlacePageLayout/Layouts/PlacePageTrackLayout.swift

```diff
@@ -0,0 +1,127 @@
+class PlacePageTrackLayout: IPlacePageLayout {
+  private var placePageData: PlacePageData
+  private var interactor: PlacePageInteractor
+  private let storyboard: UIStoryboard
+  weak var presenter: PlacePagePresenterProtocol?
+
+  lazy var bodyViewControllers: [UIViewController] = {
+    return configureViewControllers()
+  }()
+
+  var actionBar: ActionBarViewController? {
+    actionBarViewController
+  }
+
+  var navigationBar: UIViewController? {
+    placePageNavigationViewController
+  }
+
+  lazy var headerViewControllers: [UIViewController] = {
+    [headerViewController, previewViewController]
+  }()
+
+  lazy var headerViewController: PlacePageHeaderViewController = {
+    PlacePageHeaderBuilder.build(data: placePageData.previewData, delegate: interactor, headerType: .flexible)
+  }()
+
+  lazy var previewViewController: PlacePagePreviewViewController = {
+    let vc = storyboard.instantiateViewController(ofType: PlacePagePreviewViewController.self)
+    vc.placePagePreviewData = placePageData.previewData
+    return vc
+  }()
+
+  lazy var placePageNavigationViewController: PlacePageHeaderViewController = {
+    return PlacePageHeaderBuilder.build(data: placePageData.previewData, delegate: interactor, headerType: .fixed)
+  }()
+
+  lazy var bookmarkViewController: PlacePageBookmarkViewController = {
+    let vc = storyboard.instantiateViewController(ofType: PlacePageBookmarkViewController.self)
+    vc.view.isHidden = true
+    vc.delegate = interactor
+    return vc
+  }()
+
+  lazy var elevationMapViewController: ElevationProfileViewController? = {
+    guard let elevationProfileData = placePageData.trackData?.elevationProfileData else {
+      return nil
+    }
+    return ElevationProfileBuilder.build(elevationProfileData: elevationProfileData, delegate: interactor)
+  }()
+
+  lazy var actionBarViewController: ActionBarViewController = {
+    let vc = storyboard.instantiateViewController(ofType: ActionBarViewController.self)
+    vc.placePageData = placePageData
+    vc.canAddStop = MWMRouter.canAddIntermediatePoint()
+    vc.isRoutePlanning = MWMNavigationDashboardManager.shared().state != .hidden
+    vc.delegate = interactor
+    return vc
+  }()
+
+  init(interactor: PlacePageInteractor, storyboard: UIStoryboard, data: PlacePageData) {
+    self.interactor = interactor
+    self.storyboard = storyboard
+    self.placePageData = data
+  }
+
+  private func configureViewControllers() -> [UIViewController] {
+    var viewControllers = [UIViewController]()
+
+    viewControllers.append(bookmarkViewController)
+    if let trackData = placePageData.trackData {
+      bookmarkViewController.bookmarkData = .track(trackData)
```

Comment:
Should bookmarkViewController.bookmarkData be renamed to something like `bookmarkOrTrackViewController.bookmarkOrTrackData`, or would it be cleaner/simpler to separate them?

Reply:
The `PlacePageBookmarkViewController` is the VC that holds the `Edit bookmark/ Edit track` button with the description. So maybe it is worth to rename it to the smth like `PlacePageEditViewController` as the 'PlacePageInfoViewController`, PlacePageButtonViewController`

Reply:
I renamed the entities to be more clear.
`PlacePageEditBookmarkViewController` -> `PlacePageEditBookmarkOrTrackViewController`
`BookmarkData` -> `EditData`